core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -49,32 +49,45 @@ private[kafka] object LogValidator extends Logging {
                                                       sourceCodec: CompressionCodec,
                                                       targetCodec: CompressionCodec,
                                                       compactedTopic: Boolean = false,
-                                                      messageFormatVersion: Byte = RecordBatch.CURRENT_MAGIC_VALUE,
-                                                      messageTimestampType: TimestampType,
-                                                      messageTimestampDiffMaxMs: Long,
-                                                      partitionLeaderEpoch: Int = RecordBatch.UNKNOWN_PARTITION_LEADER_EPOCH): ValidationAndOffsetAssignResult = {
+                                                      magic: Byte,
```

Comment:
Sure. I had done that, but after noticing that a lot of tests relied on it, I was a bit unsure. Your suggestion settles the doubt. :)